ast/edit/task_editor.go

```diff
@@ -27,6 +27,28 @@ func GetOption(file *ast.File, name string) (ast.Expression, error) {
 	}
 }
 
+// GetAllOptions returns all options within a file
+func GetAllOptions(file *ast.File) ([]ast.Expression, error) {
+	var exps []ast.Expression
+	for _, st := range file.Body {
+		if val, ok := st.(*ast.OptionStatement); ok {
+			assign := val.Assignment
+			if va, ok := assign.(*ast.VariableAssignment); ok {
+				if ok {
+					exps = append(exps, va.Init)
+				}
+			}
+		}
+	}
+	if len(exps) == 0 {
```

Comment:
What is the case where this (and the other check for `len(...) == 0`) could happen? Is it really an error?

Reply:
This is just a way to check if there are no options or no properties when the caller checks for them. Maybe a bool would be better here than an error, but there needs to be a way to tell them that nothing was found. 

This could certainly happen. If you had a script with no options declared, then the len of the expressions array would be zero.

Reply:
Can we not return an empty list? That's a valid value, isn't it?

Reply:
From a consumer's perspective, I'd reason the empty slice would being an adequate signal that options don't exist in the query for GetAllOptions or GetAllProperties 🤔.

Reply:
🤷🏾‍♀️ I mean yeah its valid. I guess it depends on whether the use would rather check for an error or check the length of the returned array themselves.

Reply:
@jsteenb2 do you have an opinion on this?

Reply:
Oh. Some conversation overlap. That's fine. I can change this.

Reply:
If I'm using GetAll, I wouldnt' sue either. I'd just loop over whatever options/properties I am provided. GetAllOptions I dont' have a use case for, but GetAllProperties for an option, I'll use a lot of.

Reply:
yep got it. I'll go ahead and update this.

Reply:
I don't think an error is appropriate if having no options is valid. I think a good indicator is thinking about an error message like "Error: a valid thing occurred" is a good indicator that an error isn't the right thing here.

Reply:
Updated. Not returning an error anymore. Lemme know what y'all think.